### PR TITLE
feat: add core support for `database_schema` block

### PR DESCRIPTION
**Pull request by fontanierh**
Next I will do the `database` block

### CODE REVIEW

File: core/src/blocks/database_schema.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        let databases = match config {
+            Some(v) => match v.get("databases") {
+                Some(Value::Array(a)) => a
+                    .iter()
+                    .map(|v| {
+                        let workspace_id = match v.get("workspace_id") {
+                            Some(Value::String(s)) => s,
+                            _ => Err(anyhow!(err_msg.clone()))?,
+                        };
+                        let data_source_id = match v.get("data_source_id") {
```
Comment (spolu) [inline core/src/blocks/database_schema.rs:57]: This is the external data_source_id so the name of the data source here.

File: core/src/blocks/database_schema.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                            Some(Value::String(s)) => s,
+                            _ => Err(anyhow!(err_msg.clone()))?,
+                        };
+                        let database_id = match v.get("database_id") {
```
Comment (spolu) [inline core/src/blocks/database_schema.rs:61]: Where did we land on this one. That's an generated ID right?

**Review by spolu — APPROVED** [review]
LGTM

Left 1 small qestion

File: core/src/blocks/database_schema.rs (hunk 2 above)
Comment (fontanierh) [reply core/src/blocks/database_schema.rs:61]: Yes, it's generated and not provided by users. But as far as `core` is concerned it's not generated and can be anything (the SID is being generated on `front` side)

File: core/src/blocks/database_schema.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/blocks/database_schema.rs:57]: I don't understand what you mean ? Unless I am mistaken, what we call "data source name" in front is just the `datasource_id` in core (and we never call it data "name" here)

**Pull request merged**